src/com/sap/piper/tools/ToolVerifier.groovy

```diff
@@ -0,0 +1,51 @@
+package com.sap.piper.tools
+
+import com.sap.piper.FileUtils
+import com.sap.piper.Version
+
+import hudson.AbortException
+
+
+class ToolVerifier implements Serializable {
+
+    def static verifyToolHome(tool, script, configuration, environment) {
+
+        def home = ToolUtils.getToolHome(tool, script, configuration, environment)
+        if (home) { 
+            script.echo "Verifying $tool.name home."
+            FileUtils.validateDirectoryIsNotEmpty(home)
+        }
+        return home
+    }
+
+    def static verifyToolExecutable(tool, script, configuration, environment) {
+
+        def home = verifyToolHome(tool, script, configuration, environment)
+        def executable = ToolUtils.getToolExecutable(tool, script, home)
+        if (home) {
+            script.echo "Verifying $tool.name executable."
+            FileUtils.validateFile(executable)
+        }
+        return executable
+    }
+
+    def static verifyToolVersion(tool, script, configuration, environment) {
+
+        def executable = verifyToolExecutable(tool, script, configuration, environment)
+        if (tool == ToolType.MTA) executable = "$environment.JAVA_HOME/bin/java -jar $executable"
```

Comment:
It is confusing to java a java check when tool is mta. Wouldn't it be more honest to call with java directly in case we are interested in the java version? What about the mta version itself (```java -jar <mtaJar>  --version```). Don't know if this was like this before, in this case if would be OK for a refactoring PR.

Reply:
It was like this, but maybe we can improve it. We are not interested in the java version, it is to call mta.

Reply:
OK, I mis-read the code, you are right. It is about an mta version check, not about a java version check.